.circleci/config.yml

```diff
@@ -47,89 +44,98 @@ jobs:
       - restore_cache: *restore_cargo_cache
 
       - run:
-          name: Check for Errors
-          command: |
-            cargo clean
-            cargo check
-            cd fixture_setup
-            cargo check
+          name: Check Formatting of forte_core
+          command: *lint_commands
 
-      - save_cache: *save_cargo_cache
+      - run:
+          name: Check Formatting of fixture_setup
+          working_directory: fixture_setup
+          command: *lint_commands
 
       - run:
-          name: Check Formatting
-          command: |
-            cargo fmt -- --write-mode=diff
-            cd fixture_setup
-            cargo fmt -- --write-mode=diff
+          name: Compile forte_core
+          command: *compile_commands
 
       - run:
-          name: Check Linting
-          command: |
-            cargo clippy -- -D warnings
-            cd fixture_setup
-            cargo clippy -- -D warnings
+          name: Compile fixture_setup
+          working_directory: fixture_setup
+          command: *compile_commands
 
       - save_cache: *save_cargo_cache
 
-  test:
+      # The workspace stores some of the same information the cache does, but
+      # the workflow is shared across a workflow and updated on every build
+      # while the cache only updates when a lock file changes.
+      - persist_to_workspace:
+          root: .
+          paths:
+            - target
+            - fixture_setup/target
+
+  integration-test:
     docker:
-      # Primary container
       - image: fortemusic/core-build
 
-      # Secondary container (available on localhost)
-      - image: redis:4.0.6
+    environment:
+      RUST_BACKTRACE: "1"
 
     steps:
+      - attach_workspace:
+          at: /project
```

Comment:
Should this step come after `checkout`?

Reply:
It doesn't matter.

Reply:
Why not? You have to back up your reasoning somehow.

Reply:
No, you're the one who needs to back up your reasoning. I'm adding code. Take it or leave it.

Reply:
When someone reviews code and raises a question, the author needs to answer it and not just hand wave it away. I asked about this because the checkout might overwrite the workspace cache. You don't need to be so argumentative.

Reply:
The order doesn't matter whatsoever. The workspace is always attached before everything.
![image](https://user-images.githubusercontent.com/3277097/40406958-b9ad73fc-5e30-11e8-9424-97ed5d8d44af.png)